src/Library/demos/Column View/main.js

```diff
@@ -0,0 +1,123 @@
+import Gtk from "gi://Gtk";
+import Gio from "gi://Gio";
+import GObject from "gi://GObject";
+
+const column_view = workbench.builder.get_object("column_view");
+const col1 = workbench.builder.get_object("col1");
+const col2 = workbench.builder.get_object("col2");
+const col3 = workbench.builder.get_object("col3");
+
+function createFirstCol() {
+  const factory = col1.factory;
+  factory.connect("setup", (factory, list_item) => {
+    const label = new Gtk.Label({
+      margin_start: 12,
+      margin_end: 12,
+    });
+    list_item.set_child(label);
+  });
+  factory.connect("bind", (factory, list_item) => {
+    const label_widget = list_item.get_child();
+    const model_item = list_item.get_item();
+    label_widget.label = model_item.get_display_name();
+  });
+}
+
+function createSecCol() {
+  const factory = col2.factory;
+  factory.connect("setup", (factory, list_item) => {
+    const label = new Gtk.Label({
+      margin_start: 12,
+      margin_end: 12,
+    });
+    list_item.set_child(label);
+  });
+  factory.connect("bind", (factory, list_item) => {
+    const label_widget = list_item.get_child();
+    const model_item = list_item.get_item();
+    label_widget.label = model_item.get_size().toString();
+  });
+}
+
+function createThirdCol() {
+  const factory = col3.factory;
+  factory.connect("setup", (factory, list_item) => {
+    const label = new Gtk.Label({
+      margin_start: 12,
+      margin_end: 12,
+    });
+    list_item.set_child(label);
+  });
+  factory.connect("bind", (factory, list_item) => {
+    const info = list_item.item;
+    const mtime = info.get_modification_date_time();
+    const date = new Date(mtime.to_unix() * 1000);
+    const now = new Date();
+
+    const options = {};
+    if (date.getMonth() < now.getMonth() || date.getDay() < now.getDay())
+      options.dateStyle = "long";
+    else options.timeStyle = "short";
+
+    const label = list_item.child;
+    label.label = Intl.DateTimeFormat(undefined, options).format(date);
+  });
+}
+
+//Model
+const dir = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
+  "Library/demos",
+);
+
+const model = new Gtk.SingleSelection({
+  model: new Gtk.SortListModel({
+    model: new Gtk.DirectoryList({
+      file: dir,
+      attributes: "standard::*,time::modified",
+    }),
+    sorter: column_view.sorter,
+  }),
+});
```

Comment:
Cool, now that you've got a handle on this, let's break this into separate variables so it's *really* clear what's going on. This is one of the tougher things for people to grasp.

I would do something like:

```js
const data_model = new Gtk.DirectoryList({ ... });

const sort_model = new Gtk.SortListModel({ ... });

column_view.model = new Gtk.SingleSelection({ ... });
```

This way those reading the code don't have to unravel the nesting "backwards", and its really obvious what's being done.